src/mpi/coll/ineighbor_alltoall/ineighbor_alltoall_allcomm_sched_linear.c

```diff
@@ -42,11 +42,20 @@ int MPIR_Ineighbor_alltoall_allcomm_sched_linear(const void *sendbuf, MPI_Aint s
         MPIR_ERR_CHECK(mpi_errno);
     }
 
-    /* receive needs to happen in the opposite order of sends.  This
-     * is to cover the case of Cartesian graphs where the same process
-     * might be both the left and right neighbor.  In this case, we
-     * need to make sure the first message (which is from the right
-     * neighbor) is received in the last buffer. */
+    /* Cartetian graph will result in multiple edges going to the same process when it is
+     * periodic and dim is 1 or 2, in which case, both left and right neighbor points to
```

Comment:
the issue is if the size of the dimension is 1 or 2. neighbors are not necessarily left and right.

Reply:
"left and right" in a circular sense. In the case of 2 for example, saying it produces two edges both pointing to the neighboring process leaves one wondering why. But that maybe just me. Feel free to suggest better wording.